ui/apps/platform/src/Containers/NetworkGraph/components/NetworkBreadcrumbs.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useCallback, ChangeEvent } from 'react';
+import {
+    Breadcrumb,
+    BreadcrumbItem,
+    Select,
+    SelectOption,
+    SelectVariant,
+} from '@patternfly/react-core';
+
+import { Cluster } from 'types/cluster.proto';
+import useURLSearch from 'hooks/useURLSearch';
+import useFetchClusterNamespaces from 'hooks/useFetchClusterNamespaces';
+import { ClusterIcon, NamespaceIcon } from '../common/NetworkGraphIcons';
+import getScopeHierarchy from '../utils/getScopeHierarchy';
+
+function filterElementsWithValueProp(
+    filterValue: string,
+    elements: React.ReactElement[] | undefined
+): React.ReactElement[] | undefined {
+    if (filterValue === '' || elements === undefined) {
+        return elements;
+    }
+
+    return elements.filter((reactElement) =>
+        reactElement.props.value?.toLowerCase().includes(filterValue.toLowerCase())
+    );
+}
+
+type NetworkBreadcrumbsProps = {
+    clusters: Cluster[];
+};
+
+function NetworkBreadcrumbs({ clusters = [] }: NetworkBreadcrumbsProps) {
+    const [isClusterOpen, setIsClusterOpen] = React.useState(false);
+    const [isNamespaceOpen, setIsNamespaceOpen] = React.useState(false);
+
+    const { searchFilter, setSearchFilter } = useURLSearch();
+
+    const {
+        cluster: selectedClusterName,
+        namespaces: selectedNamespaces,
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        deployments: deploymentsFromUrl,
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        remainingQuery,
+    } = getScopeHierarchy(searchFilter);
+    const selectedClusterId =
+        clusters.find((cluster) => cluster.name === selectedClusterName)?.id || '';
+    const { loading, error, namespaces } = useFetchClusterNamespaces(selectedClusterId);
+
+    const onClusterToggle = (newClusterOpenValue) => setIsClusterOpen(newClusterOpenValue);
+
+    const onNamespaceToggle = (newNamespaceOpenValue: boolean) =>
+        setIsNamespaceOpen(newNamespaceOpenValue);
+
+    const onClusterSelect = (_, value) => {
+        setIsClusterOpen((prevIsOpen: boolean) => !prevIsOpen);
+
+        if (value !== selectedClusterName) {
+            const modifiedSearchObject = { ...searchFilter };
+            modifiedSearchObject.Cluster = value;
+            setSearchFilter(modifiedSearchObject);
+        }
+    };
+
+    const onFilterNamespaces = useCallback(
+        (e: ChangeEvent<HTMLInputElement> | null, filterValue: string) =>
+            filterElementsWithValueProp(
+                filterValue,
+                namespaces.map((filteredNS) => (
+                    <SelectOption key={filteredNS} value={filteredNS}>
```

Comment:
A little farther down I noticed something similar where we take the namespaces and map them to get an array of SelectOptions. Can we re-use that logic here too?

Reply:
Also nit: I think calling it `namespace` rather than `filteredNS` makes more sense. I was a little confused with the use of that term